source/DasBlog.Web.UI/Controllers/AccountController.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using DasBlog.Web.UI.Models.AccountViewModels;
+using DasBlog.Web.UI.Models.Identity;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+
+namespace DasBlog.Web.UI.Controllers
+{
+	[Authorize]
+	public class AccountController : Controller
+	{
+		private const string KEY_RETURNURL = "ReturnUrl";
+		private readonly UserManager<DasBlogUser> _userManager;
+		private readonly SignInManager<DasBlogUser> _signInManager;
+		private readonly IMapper _mapper;
+		private readonly ILogger _logger;
+
+		public AccountController(
+			UserManager<DasBlogUser> userManager,
+			SignInManager<DasBlogUser> signInManager,
+			IMapper mapper,
+			ILoggerFactory loggerFactory)
+		{
+			_userManager = userManager;
+			_signInManager = signInManager;
+			_mapper = mapper;
+			_logger = loggerFactory.CreateLogger<AccountController>();
```

Comment:
This is cool, I need to do some reading and start integrating this concept everywhere